h/api/models/elastic.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from annotator import annotation
+from annotator import document
+
+from h._compat import text_type
+
+
+class Annotation(annotation.Annotation):
+    @property
+    def uri(self):
+        """Return this annotation's URI or an empty string.
+
+        The uri is escaped and safe to be rendered.
+
+        The uri is a Markup object so it won't be double-escaped.
+
+        """
+        uri_ = self.get("uri")
+        if uri_:
+            # Convert non-string URIs into strings.
+            # If the URI is already a unicode string this will do nothing.
+            # We're assuming that URI cannot be a byte string.
+            return text_type(uri_)
+        else:
+            return ""
+
+    @property
+    def parent(self):
+        """
+        Return the thread parent of this annotation, if it exists.
+        """
+        if 'references' not in self:
```

Comment:
It could. I'm not sure that would be more readable. And you'd still have to check if it was empty or not to preserve behaviour.

Also, in case it's not clear: this code is unchanged. I'm just moving it.